From f0bad617ba038e6d8aae9a474728920e46ed9ded Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Thu, 31 Mar 2016 16:02:38 -0400
Subject: [PATCH 0996/1110] drm/amd/dal: Fix the typo in apply_ctx_to_hw reset
 logic

This fixes intermittent black screen on hotpluging second
display on linux

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index e5cb1aa..df52740 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1194,7 +1194,7 @@ static enum dc_status apply_ctx_to_hw(
 		 * since bios does optimization and doesn't apply if changing
 		 * PHY when not already disabled.
 		 */
-		if (pipe_ctx_old->stream && pipe_ctx_old->stream != pipe_ctx->stream)
+		if (pipe_ctx_old->stream && !pipe_ctx->stream)
 			reset_single_pipe_hw_ctx(
 				dc, pipe_ctx_old, &dc->current_context);
 		else if (pipe_ctx_old->clock_source != pipe_ctx->clock_source)
-- 
2.7.4

